src/format.ts

```diff
@@ -110,8 +120,117 @@ function checkFormat(srcFiles: string[], baseArgs: string[]): Promise<boolean> {
     out.on('data', (data: Buffer) => {
       output += data;
     });
+
     out.on('end', () => {
-      resolve(output.indexOf('<replacement ') === -1 ? true : false);
+      findFormatErrorLines(output, options).then(() => {
+        resolve(output.indexOf('<replacement ') === -1 ? true : false);
+      });
+    });
+  });
+}
+
+/**
+ * Parses through xml string for the replacement offsets and lengths. Uses those
+ * values to locate the formatting error lines.
+ *
+ * @param output xml string
+ * @param options
+ */
+async function findFormatErrorLines(output: string, options: Options) {
+  const files = output.split('<?xml version=\'1.0\'?>\n');
+
+  for (let i = 1; i < files.length; i++) {
+    const formatErr = {
+      offset: output.match(/(?<=offset=\')(\d+)(?=\')/g),
+      length: output.match(/(?<=length=\')(\d+)(?=\')/g),
+      fix: output.match(/(?<=length=\'\d+\'>)(.*)(?=<\/replacement>)/g)
+    };
+
+    if (formatErr.length === null || formatErr.offset === null ||
+        formatErr.fix === null) {
+      return;
+    }
+
+    for (let j = 0; j < formatErr.fix.length; j++) {
+      formatErr.fix[j] = entities.decodeXML(formatErr.fix[j]);
+    }
+
+    const read = promisify(fs.readFile);
+    const argNum = 3;
+    const file = process.argv[argNum + i - 1];
```

Comment:
Is it guaranteed that the files will be in `process.argv`? I believe the list of arguments passed to `clang-format` come from a different source?

Reply:
Also, could you add an inline comment explaining why it's safe to use `i` to index into a different array than `files`?

Reply:
I just realized that too... I used process.argv because it provides the file name, but my algorithm makes the assumption that the user passes the absolute path of a file. The xml string does not provide the file names, so how can I get them?

Reply:
fixed!